### DISCUSSION TITLE
Where is the META-INF.services file in the project?

### DISCUSSION THREAD

**Discussion by SniperWoods** [apache/dolphinscheduler#12554] [Q&A] [answered]
DophinScheduler uses java spi, why can not find files like META-INF.services/org.apache.dolphinscheduler.plugin.task.api.TaskChannelFactory in the project?

**Comment by DarkAssassinator** [comment, marked as answer]
hi @SniperWoods, because DS use Google `@AutoService`, it will helps us with generating SPI configuration files.

**Reply by SniperWoods** [reply]
Appreciate it
